### PR TITLE
E2e/parking sensors add synapse

### LINKED ISSUE #385: Add a Synapse module for bicep deployment to include Synapse account and workspace
Add the following to the bicep modules:
- Spark Pool
- Log Analytics
- Role Assignments
- Synapse workspace
- Synapse Storage

### LINKED ISSUE #389: Add a Log Analytics module for bicep deployment (including loging kusto query). Configure spark pool in synapse to point to the Log Analytics deployed workspace.
## Description
Add a Log Analytics module for bicep deployment (including logging kusto query). Configure spark pool in synapse to point to the Log Analytics deployed workspace.

### LINKED ISSUE #403: CI: Deploy whl packages and requirements.txt as part of the initial deployment

### PR DESCRIPTION
**Pull request by ydaponte**
# Type of PR
- Documentation changes
- Code changes

## Purpose

- Adds the deployment of a Synapse workspace and a Synapse Spark Pool as a baseline to port the parking sensor sample to synapse
- Adds the deployment of a Log Analytics workspace with saved kusto query to analyze notebooks logs later
- Automates update of the synapse workspace to contain needed whl packages located on /synapse/libs folder
- Configures spark pool to include whl packages
- Configures spark pool to point to requirements.txt that is located on synapse/config

## Does this introduce a breaking change? If yes, details on what can break
Not completely a breaking change. On the main.bicep file I'm referencing to a storage_v2.bicep module instead of the previous storage.bicep version. Reason is we are deploying a second storage account to be the default sa for the synapse workspace.

## Author pre-publish checklist

- [ X] No PII in logs
- [ X] Made corresponding changes to the documentation

## Validation steps

- Deployed the e2e sample with the changes,  full sample deployment was successful. 
- Access to Synapse workspace in all environments and check spark pool configuration settings and package settings for requirements.txt
- Check if Log Analytics workspace (Id and Key) are correctly referenced on the spark pool configuration for each environment
- On the Azure Portal, navigate to the deployed synapse workspace/spark pool/Packages and check if Requirement files is filled 
with requirements.txt 
- On the Azure Portal, navigate to the deployed synapse workspace/spark pool/Packages and check if Workspace packages are filled with the following file: ddo_transform-localdev-py2.py3-none-any.whl.

## Issues Closed or Referenced

- Closes #385 
- Closes #389 
- Closes #403

### CODE REVIEW

File: e2e_samples/parking_sensors/infrastructure/modules/storage_v2.bicep [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      keySource: 'Microsoft.Storage'
+    }
+    accessTier: 'Hot'
+  }
+}
+
+resource storageFileSystem2 'Microsoft.Storage/storageAccounts/blobServices/containers@2021-04-01' = {
+  name: '${project}st2${env}${deployment_id}/default/container001'
+  properties: {
+    publicAccess: 'Container'
```
Comment (devlace) [inline e2e_samples/parking_sensors/infrastructure/modules/storage_v2.bicep:95]: Is the publicAccess necessary? I got some compliance warnings that I have a unsecure storage container after deploying the solution..

File: e2e_samples/parking_sensors/infrastructure/modules/synapse.bicep [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  name: guid(resourceGroup().id, resourceId('Microsoft.Storage/storageAccounts', synStorage.name))
+  properties: {
+    principalId: synapseWorkspace.identity.principalId
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe')
+    principalType: 'ServicePrincipal'
+  }
+  scope: synStorage
+}
+
+resource roleAssignment2 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
```
Comment (devlace) [inline e2e_samples/parking_sensors/infrastructure/modules/synapse.bicep:142]: Just copy-pasting my previous comment in the older PR #386 :)

Could we rename to something more meaningful like `roleAssignemntSynStorage` ?

Also would be good to set the roleDefinitionId to variables names that describe the role so its much more readable than just a GUID.  
Something like this: https://github.com/Azure-Samples/modern-data-warehouse-dataops/blob/226e0443a009a145e8583de574b973954d82d8c7/e2e_samples/parking_sensors/infrastructure/modules/storage.bicep#L13

**Review by devlace — CHANGES_REQUESTED** [review]

File: e2e_samples/parking_sensors/infrastructure/modules/storage_v2.bicep (hunk 1 above)
Comment (ydaponte) [reply e2e_samples/parking_sensors/infrastructure/modules/storage_v2.bicep:95]: Not at all. Good catch, I've corrected it - will push the changes.

File: e2e_samples/parking_sensors/infrastructure/modules/synapse.bicep (hunk 2 above)
Comment (ydaponte) [reply e2e_samples/parking_sensors/infrastructure/modules/synapse.bicep:142]: Both corrected. Tks Lace.
I've also ran into an issue with your main storage account deployment. I've added the "principalType" property  and that solved the problem.

**Review by devlace — APPROVED** [review]

**Pull request merged**